lib/restforce/concerns/streaming.rb

```diff
@@ -30,6 +31,65 @@ def faye
           client.bind 'transport:up' do
             Restforce.log "[COMETD UP]"
           end
+
+          client.add_extension ReplayExtension.new(replay_handlers)
+        end
+      end
+
+      def replay_handlers
+        @_replay_handlers ||= {}
+      end
+
+      class ReplayExtension
+        def initialize(replay_handlers)
+          @replay_handlers = replay_handlers
+        end
+
+        def incoming(message, callback)
+          callback.call(message).tap do
+            channel = message.fetch('channel').gsub('/topic/', '')
+            replay_id = message.fetch('data', {}).fetch('event', {})['replayId']
+
+            handler = @replay_handlers[channel]
+            if !replay_id.nil? && !handler.nil? && handler.respond_to?(:[]=)
```

Comment:
I think there is a bug here. If the replay_id is initialized with a scalar value, handler.respond_to?(:[]=) will be false and the replay_id is never updated. The "remember last replay_id" assignment logic should probably mirror what the replay_id(channel) method is doing.